lib/backend/kubernetes/kubernetes.go

```diff
@@ -61,12 +65,16 @@ type Config struct {
 	// Namespace is the Agent's namespace
 	// Field is required
 	Namespace string
-	// SecretName is unique secret per agent where state and identity will be stored.
+	// SecretName uniquely identifies the secret. Conventionally this will be set
```

Comment:
What is the secret though? The godoc refers to the secret as if the reader should already know what the secret is for.